Lib/shutil.py

```diff
@@ -1449,6 +1452,16 @@ def _access_check(fn, mode):
             and not os.path.isdir(fn))
 
 
+def _win_path_needs_curdir(cmd, mode):
+    """
+    On Windows, we can use NeedCurrentDirectoryForExePath to figure out
+    if we should add the cwd to PATH when searching for executables if
+    the mode is executable.
+    """
+    return (not (mode & os.X_OK)) or _winapi.NeedCurrentDirectoryForExePath(
+                cmd if isinstance(cmd, str) else os.fsdecode(cmd))
```

Comment:
```suggestion
                os.fsdecode(cmd))
```

This is a no-op on `str`, so just call it.